### PR TITLE
A few more integration tests

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

File: Library/Homebrew/test/test_integration_cmds.rb [hunk 1]
```diff
@@ -214,26 +219,39 @@ class Bar < Formula
...
+        url "https://example.com/foo-1.0.tar.gz"
+      end
+    EOS
+    findutils_file.write <<-EOS.undent
+      class Findutils < Formula
+        url "https://example.com/foo-1.0.tar.gz"
+      end
+    EOS
+    assert_match "This is an integration test",
+                 cmd_fail("doctor")
```
Comment (xu-cheng) [inline Library/Homebrew/test/test_integration_cmds.rb]: I think we could run only `check_intergration_test`. Rest of actual checks should be handled by unittest.

Comment (mikemcquaid) [inline Library/Homebrew/test/test_integration_cmds.rb]: Personally I'm :+1: as-is.

Comment (xu-cheng) [inline Library/Homebrew/test/test_integration_cmds.rb]: My concerns are:
* brew doctor itself does take some time. Especially for check Homebrew outdate 
* It's already be run `test-bot`. 
* The result will be dependent to running environment, which is probably not what we want inside test case.

Comment (mikemcquaid) [inline Library/Homebrew/test/test_integration_cmds.rb]: Cool, that's fine, let's go @xu-cheng's way although I would like it do be done without adding dead/test-only methods to `doctor` or `diagnostic` code

Comment (bfontaine) [inline Library/Homebrew/test/test_integration_cmds.rb]: Would something like that work for you @mikemcquaid?
```rb
      if ENV["HOMEBREW_INTEGRATION_TEST"]
        def check_integration_test
          "This is an integration test"
        end
      end
```

Comment (xu-cheng) [inline Library/Homebrew/test/test_integration_cmds.rb]: We could tweak `cmd_output` a bit to allow us inject code using Ruby options, `-I` and `-r`.

File: Library/Homebrew/diagnostic.rb [hunk 2]
```diff
@@ -1336,6 +1336,13 @@ def check_for_external_cmd_name_conflict
         s
       end
 
+      def check_integration_test
+        if ENV["HOMEBREW_INTEGRATION_TEST"]; <<-EOS.undent
+            This is an integration test
+          EOS
+        end
+      end
```
Comment (mikemcquaid) [inline Library/Homebrew/diagnostic.rb]: Would be good if this can live in `test/` somewhere instead.

Comment (bfontaine) [inline Library/Homebrew/diagnostic.rb]: I moved it in `test/lib/integration_mocks.rb`.

Comment (mikemcquaid) [inline Library/Homebrew/diagnostic.rb]: :+1:

**Pull request closed without merging**